common/src/main/java/com/microsoft/identity/common/internal/controllers/ExceptionAdapter.java

```diff
@@ -188,24 +198,143 @@ public static ServiceException getExceptionFromTokenErrorResponse(@NonNull final
+        final String methodName = ":getUIdFromHomeAccountId";
+        final String DELIMITER_TENANTED_USER_ID = ".";
+        final int EXPECTED_ARGS_LEN = 2;
+        final int INDEX_USER_ID = 0;
+
+        if (!TextUtils.isEmpty(homeAccountId)) {
+            final String[] homeAccountIdSplit = homeAccountId.split(
+                    Pattern.quote(DELIMITER_TENANTED_USER_ID)
+            );
+
+            if (homeAccountIdSplit.length == EXPECTED_ARGS_LEN) {
+                com.microsoft.identity.common.internal.logging.Logger.info(TAG + methodName,
+                        "Home account id is tenanted, returning uid "
+                );
+                return homeAccountIdSplit[INDEX_USER_ID];
+            } else if (homeAccountIdSplit.length == 1) {
+                com.microsoft.identity.common.internal.logging.Logger.info(TAG + methodName,
+                        "Home account id not tenanted, it's the uid added by v1 broker "
+                );
+                return homeAccountIdSplit[INDEX_USER_ID];
+            }
+        }
+
+        com.microsoft.identity.common.internal.logging.Logger.warn(TAG + methodName,
+                "Home Account id doesn't have uid or tenant id information, returning null "
+        );
+
+        return null;
+    }
+
+    private static void setIntuneExceptionProperties(
+            @NonNull final IntuneAppProtectionPolicyRequiredException exception,
+            @NonNull final BrokerSilentTokenCommandParameters requestParameters) {
+
+        com.microsoft.identity.common.internal.logging.Logger.info(TAG, "Setting properties to IntuneAppProtectionPolicyRequiredException ");
+
+        final String upn = (requestParameters.getAccountManagerAccount() != null) ?
+                requestParameters.getAccountManagerAccount().name :
+                requestParameters.getLoginHint();
+        exception.setAccountUpn(upn);
+
+        String uId = requestParameters.getLocalAccountId();
+
+        if (TextUtils.isEmpty(uId)) {
+            com.microsoft.identity.common.internal.logging.Logger.info(TAG, "Local account id is empty, attempting get user id from home account id");
+            uId = getUIdFromHomeAccountId(
+                    requestParameters.getHomeAccountId()
+            );
+        }
+
+        exception.setAccountUserId(uId);
+
+        final Authority authority = requestParameters.getAuthority();
+        exception.setAuthorityUrl(authority.getAuthorityURL().toString());
+
+        final String homeAccountId = requestParameters.getHomeAccountId();
+        String tenantId = null;
+
+        if (homeAccountId != null) {
+            final Pair<String, String> tenantInfo = StringUtil.getTenantInfo(homeAccountId);
+            tenantId = tenantInfo.second;
+        }
+
+        if (TextUtils.isEmpty(tenantId) && authority instanceof AzureActiveDirectoryAuthority) {
+            tenantId = ((AzureActiveDirectoryAuthority) authority).mAudience.getTenantId();
+
+        }
```

Comment:
Since it looks like we can get the tenant id here from multiple different places, would logging where we got it help solve any problems?

Reply:
No immediate problems related to what we're trying to fix here, but we can certainly review this all up.  I think it's worthwhile to do a code walkthrough in our weekly meeting to identify similar potentially pervasive issues.  I feel like there was some WPJ code still hanging around in here.